### PR TITLE
readability enhancement for cj controller unit test

### LINKED ISSUE #107633: readability enhancement on unit test of cron job controller
### What happened?

in TestControllerV2SyncCronJob, there are many listing the properties that don't matter (like nil, false). and many constants F and T, which is unnecessary.

### What did you expect to happen?

As per @alculquicondor 's request, We shouldn't be listing the properties that don't matter (like nil, false). And get rid of constants F and T

### How can we reproduce it (as minimally and precisely as possible)?

check pkg/controller/cronjob/cronjob_controllerv2_test.go

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
v1.21.3 and later
```

</details>

### Cloud provider

<details>
None
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by jlsong01**
#### What type of PR is this?
/kind cleanup
/sig apps

#### What this PR does / why we need it:
in TestControllerV2SyncCronJob, there are many listing the properties that don't matter (like nil, false). and many constants F and T, which is unnecessary.

We shouldn't be listing the properties that don't matter (like nil, false). And get rid of constants F and T

#### Which issue(s) this PR fixes:

Fixes #107633

#### Special notes for your reviewer:
the pr only remove the listing the properities with default value(like nil, false, 0), and get rid of T and F.
The other properities value remains untouched

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: pkg/controller/cronjob/cronjob_controllerv2_test.go [hunk 1]
```diff
@@ -51,11 +51,9 @@ var (
 	// schedule is hourly on the hour
 	onTheHour = "0 * * * ?"
 
-	A = batchv1.AllowConcurrent
-	f = batchv1.ForbidConcurrent
-	R = batchv1.ReplaceConcurrent
-	T = true
-	F = false
+	AllowConcurrent   = batchv1.AllowConcurrent
```
Comment (alculquicondor) [inline pkg/controller/cronjob/cronjob_controllerv2_test.go:54]: prefer to get rid of these constants as well

File: pkg/controller/cronjob/cronjob_controllerv2_test.go [hunk 2]
```diff
@@ -190,70 +188,573 @@ func TestControllerV2SyncCronJob(t *testing.T) {
 		jobPresentInCJActiveStatus bool
```
Comment (alculquicondor) [inline pkg/controller/cronjob/cronjob_controllerv2_test.go:188]: you can move these variable into "cj status"?

**Review by alculquicondor — COMMENTED** [review]
Thanks

**Comment by alculquicondor** [comment]
Can I ask how you did the conversion? Did you do it all manually or did you get the assistance from an IDE?

It's almost impossible to review this in detail, so I want to know what is the likelihood that you did a mistake when doing the expansion?

/ok-to-test

File: pkg/controller/cronjob/cronjob_controllerv2_test.go (hunk 2 above)
Comment (jlsong01) [reply pkg/controller/cronjob/cronjob_controllerv2_test.go]: done

File: pkg/controller/cronjob/cronjob_controllerv2_test.go (hunk 1 above)
Comment (jlsong01) [reply pkg/controller/cronjob/cronjob_controllerv2_test.go]: done

**Comment by jlsong01** [comment]
> Can I ask how you did the conversion? Did you do it all manually or did you get the assistance from an IDE?
> 
> It's almost impossible to review this in detail, so I want to know what is the likelihood that you did a mistake when doing the expansion?
> 
> /ok-to-test

Yeah, I did it with the help from an IDE named JetBrains Goland.

Yes, seems it's impossible to review one by one as huge changes here.

I think IDE Goland is very power which can show each property name along with the property value, this feature helps me a lot. 
Also, I carefully removed the default value while keep non-default vaule untouched. 
Finally, I have locally run the unit test  again after all change done, and it passed. in case of any mistake, the unit test will complain, but in fact it not.
So, I think this would probably ok.

**Comment by alculquicondor** [comment]
GoLand has a feature to automatically expand structs. But in any case, I'll wait for the tests to run and tag it.

**Comment by alculquicondor** [comment]
/retest
/lgtm
/assign @kow3ns
for approval

**Comment by jlsong01** [comment]
@kow3ns Hey, Kenneth, we have another PR(https://github.com/kubernetes/kubernetes/pull/107470) which is dependent on the current one. so, could you please process it with higher priority. Thanks a lot.

**Review by janetkuo — APPROVED** [review]

**Pull request merged**